crates/hir-def/src/resolver.rs

```diff
@@ -163,22 +169,33 @@ impl Resolver {
         self.resolve_module_path(db, path, BuiltinShadowMode::Module)
     }
 
-    pub fn resolve_path_in_type_ns(
-        &self,
-        db: &dyn DefDatabase,
-        path: &Path,
-    ) -> Option<(TypeNs, Option<usize>, Option<ImportOrExternCrate>)> {
+    pub fn resolve_path_in_type_ns<'a>(
+        &'a self,
+        db: &'a dyn DefDatabase,
+        path: &'a Path,
+    ) -> impl Iterator<Item = (ModuleOrTypeNs, Option<usize>, Option<ImportOrExternCrate>)> + 'a
+    {
```

Comment:
Given edition 2024 new capture rules there shouldn't be a need to define the lifetime `'a`.